scripts/generate_driver_wrappers.py

```diff
@@ -22,52 +22,117 @@
 
 import sys
 import os
+import json
+from typing import Tuple
 import argparse
+import jsonschema
+from jsonschema import validate
 import jinja2
 from mbedtls_dev import build_tree
 
-def render(template_path: str) -> str:
+def render(template_path: str, driver_jsoncontext: list) -> str:
     """
-    Render template from the input file.
+    Render template from the input file and driver JSON.
     """
     environment = jinja2.Environment(
         loader=jinja2.FileSystemLoader(os.path.dirname(template_path)),
         keep_trailing_newline=True)
     template = environment.get_template(os.path.basename(template_path))
 
-    return template.render()
+    return template.render(drivers=driver_jsoncontext)
 
-def generate_driver_wrapper_file(mbedtls_root: str, output_dir: str) -> None:
+
+def generate_driver_wrapper_file(template_dir: str, \
+                                 output_dir: str, driver_jsoncontext: list) -> None:
     """
     Generate the file psa_crypto_driver_wrapper.c.
     """
     driver_wrapper_template_filename = \
-        os.path.join(mbedtls_root, \
-        "scripts/data_files/driver_templates/psa_crypto_driver_wrappers.c.jinja")
+        os.path.join(template_dir, "psa_crypto_driver_wrappers.c.jinja")
 
-    result = render(driver_wrapper_template_filename)
+    result = render(driver_wrapper_template_filename, driver_jsoncontext)
 
     with open(os.path.join(output_dir, "psa_crypto_driver_wrappers.c"), 'w') as out_file:
         out_file.write(result)
 
+
+def validate_json(driverjson_data: list, driverschema: list) -> bool:
+    """
+    Validate the Driver JSON against schema
+    """
+    try:
+        validate(instance=driverjson_data, schema=driverschema)
+    except jsonschema.exceptions.ValidationError as err:
+        print(err)
+        print("The driver JSON data is InValid")
+        return False
+
+    return True
+
+def merge_driverjsonfiles(mbedtls_root: str, json_directory: str, \
```

Comment:
Since this function reads and validates the JSON files, and not just merges them, I think a better name would be `read_driver_descriptions`.